### PR TITLE
[llama] Enable flash attention path

### PR DESCRIPTION
**Pull request by rsuderman**
We can use `torch.nn.functional.scaled_dot_product_attention` to target hte `iree` flash attention path. Substitution is a numerically identical behavior that hits `linalg_ext.attention`.

### CODE REVIEW

**Comment by rsuderman** [comment]
Kunwar is going to work on causal / attention mask for `linalg_ext`. Once that lands we can land the flash attention change.

**Review by stellaraccident — APPROVED** [review]

**Comment by stellaraccident** [comment]
We can land this, but I'll probably also want to make attention be an inference op so we can interface it with the sharding stuff. Fine to sequence that after this, since I'll be working on sharding the regular matmuls in the model for the rest of the week.
